### ISSUE TITLE
module scope fixture invoked multiple times if using test node specifiers for same module

### ISSUE THREAD

**Issue by pytestbot** [pytest-dev/pytest#723]
Originally reported by: **Sandeep Kasargod (BitBucket: [kasargod](http://bitbucket.org/kasargod), GitHub: [kasargod](http://github.com/kasargod))**

---

Using Version 2.6.4

The attachment mytest.py defines a module scoped fixture used by two tests.
Running the tests with method 1 or 2 results in the fixture setup once only as expected. However if I use method 3, where I specify the two test nodes explicitly, looks like they are always treated as two separate modules. 
Is this expected behavior or should pytest do module coalescing when test collection takes place ?
Also methods 1&2 display 'collected 2 items', method 3 displays 'collected 6 items'. I could understand 2 or 4, how does one account for 6 ?

The reason I want to use method 3 is to be able to use a text file that specifies the tests using the test node syntax as in method 3
1. py.test  mytest.py -v -s
2. py.test  mytest.py -v -s -k 'test_1 or test_2'
3. py.test  mytest.py::test_1 mytest.py::test_2 -v -s

Output for # 1 or 2

```
#!python
collected 2 items 

mytest.py::test_1 
> module_fixture_start

TEST1
PASSED
mytest.py::test_2 
TEST2
PASSED
< module_fixture_cleanup
```

Output for # 3

```
#!python
collected 6 items 

mytest.py::test_1 
> module_fixture_start

TEST1
PASSED
< module_fixture_cleanup

mytest.py::test_2 
> module_fixture_start

TEST2
PASSED
< module_fixture_cleanup

```

platform linux2 -- Python 2.7.5 -- py-1.4.26 -- pytest-2.6.4 -- /home/test/python_virtualenv/bin/python

---
- Bitbucket: https://bitbucket.org/pytest-dev/pytest/issue/723

**Comment by RonnyPfannschmidt** [comment]
@kasargod could you perhaps provide that example file? while it seems we could easily replicate what you created, we much prefer having the exactly same thing in order to avoid accidental differences created by the bias of different experiences

**Comment by RonnyPfannschmidt** [comment]
oh, wait, sorry for the buzz, the attachment was skipped in the issuee converting

**Comment by bluetech** [comment]
This was fixed by fedc78522bd43e82d8b6c4f89df2b3a6c8e898ea.
